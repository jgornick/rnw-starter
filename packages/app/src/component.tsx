import React from 'react'
import { HelloWorld } from '@jgornick/rnw-starter-hello-world'
import { Text, View } from 'react-native'

export const App = () => <View><Text>App</Text><HelloWorld /></View>
