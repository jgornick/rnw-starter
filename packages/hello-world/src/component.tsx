import React from 'react'
import { View, Text } from 'react-native'

export const HelloWorld = () => <View><Text>Hello Native World!</Text></View>
