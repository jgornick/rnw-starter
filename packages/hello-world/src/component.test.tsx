import { shallow } from 'enzyme'
import React from 'react'
import { Text } from 'react-native'

import { HelloWorld } from './component'

describe('<HelloWorld>', () => {
  const createWrapper = () => shallow<typeof HelloWorld>(
    <HelloWorld />
  )

  type WrapperType = ReturnType<typeof createWrapper>

  let wrapper: WrapperType

  beforeEach(() => {
    wrapper = createWrapper()
  })

  it('should render "Hello Native World!"', () => {
    expect(wrapper.find(Text).children()).toHaveText('Hello Native World!')
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
