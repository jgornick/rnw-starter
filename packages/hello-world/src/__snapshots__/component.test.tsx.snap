// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HelloWorld> should match snapshot 1`] = `
<View>
  <Text>
    Hello Native World!
  </Text>
</View>
`;
